components/scream/src/share/tests/vertical_interp_tests.cpp

```diff
@@ -11,46 +11,192 @@ using Spack = SmallPack<Real>;
+    for (int i=0; i<4; i++){
+      if (perturb == 0){
+        //The 3rd and 4th test are redundant in the case of the
+        //same target levels as source levels
+        if(i > 1){ break;}
+        //In this case the target levels are the same as the source levels
+        n_layers_tgt[i] = n_layers_src[i];
+
+        printf ("      -> Testing %d source layers, %d target layers\n",
+                n_layers_src[i],n_layers_tgt[i]);
+      }
+      else{
+        printf ("      -> Testing %d source layers, %d target layers\n",
+                n_layers_src[i],n_layers_tgt[i]);
+      }
+      auto npacks_src = ekat::PackInfo<N>::num_packs(n_layers_src[i]);
+      auto npacks_tgt = ekat::PackInfo<N>::num_packs(n_layers_tgt[i]);
+      auto p_tgt = view_1d<Pack<Real,N>>("",npacks_tgt);
+      auto p_tgt_s = Kokkos::create_mirror_view(ekat::scalarize(p_tgt));
+      auto tmp_src = view_2d<Pack<Real,N>>("",2,npacks_src);
+      auto tmp_src_s = Kokkos::create_mirror_view(ekat::scalarize(tmp_src));
+      auto p_src = view_2d<Pack<Real,N>>("",2,npacks_src);
+      auto p_src_s = Kokkos::create_mirror_view(ekat::scalarize(p_src));
+      auto out = view_2d<Pack<Real,N>>("",2,npacks_tgt);
+      auto out_s = Kokkos::create_mirror_view(ekat::scalarize(out));
+      auto mask = view_2d<Mask<N>>("",2,npacks_tgt);
+  
+      //Set target levels    
+      for (int lev=0; lev<(n_layers_tgt[i]-1); lev++){
+        p_tgt_s(lev) = 2*lev+perturb;
+      } 
+      //Set source levels and source input (tmp_src)
+      for (int col=0; col<2; col++){
+        for (int lev=0; lev<n_layers_src[i]; lev++){
+          p_src_s(col,lev) = 2.*lev;
+          tmp_src_s(col,lev) = 100.*(col+1) + 2.*lev;
+        }//end of looping over levels
+      }//end of looping over columns
+  
+      perform_vertical_interpolation(p_src,
+                                     p_tgt,
+                                     tmp_src,
+                                     out,
+                                     mask,
+                                     n_layers_src[i],
+                                     n_layers_tgt[i]);
+
+      //Check that output of interpolation is as expected
+      for(int col=0; col<2; col++){
+        for(int lev=0; lev<(n_layers_tgt[i]-1); lev++){
+          if (perturb == 0){
+            REQUIRE(out_s(col,lev) == tmp_src_s(col,lev));
+          }
+          else{
+            auto avg_val = (tmp_src_s(col,lev+1)+tmp_src_s(col,lev))/2.;
+            REQUIRE(out_s(col,lev) == avg_val);
+  	  }
+        }//end of looping over levels
+      }//end of looping over columns
+  
+      //Take subview and run through the 1d interpolator function 
+      //and make sure get same thing back
+      ekat::LinInterp<Real,N> vert_interp(2,n_layers_src[i],n_layers_tgt[i]);
+      const int num_vert_packs = p_tgt.extent(0);
+      const auto policy = ESU::get_default_team_policy(1, num_vert_packs);
+      Kokkos::parallel_for("scream_vert_interp_setup_loop", policy,
```

Comment:
As a procaution, I would fill `out` with garbage before running this second test. This would prevent seeing the correct results from the first test above in case for some reason this loop doesn't run.

Reply:
Same for `mask`.

Reply:
I made new variables instead of using the ones from the first test, so this should be resolved